vscode/interop/interop.ml

```diff
@@ -117,6 +125,31 @@ module Js = struct
     type t = T.t option [@@js]
   end
 
+  module Result (Ok : T) (Error : T) = struct
+    type t = (Ok.t, Error.t) result
+
+    type js_result =
+      { case : string
+      ; ok : Ojs.t
+      ; error : Ojs.t
+      }
+    [@@js]
+
+    let t_of_js js_val =
+      match js_result_of_js js_val with
+      | { case = "ok"; ok; _ } -> Ok ([%js.to: Ok.t] ok)
+      | { case = "error"; error; _ } -> Error ([%js.to: Error.t] error)
+      | _ -> assert false
```

Comment:
I changed this implementation because the previous version would reach the `assert false` when the `Ok` type is `unit`:
```
rejected promise not handled within 1 second: 0,248,Assert_failure,-10,0,vscode/interop/interop.ml,142,13
```

Because `unit` was turned into `undefined`, the function couldn't determine the difference between a unit value or the absence of one.

Checking a tag like I do here should avoid the problem.